### ISSUE TITLE
Unpack pptx file

### ISSUE THREAD

**Issue by aetonsi** [TomWildenhain/pptcc#6]
Hi,
i'd suggest unpacking the pptx file to have better versioning and allow working on the same file as granularly as possible.
Also, it would allow batch editing of slides by editing the xml files.
I'm preparing a PR right now for this.

**Comment by rugk** [comment]
How did you unpack it? Just unzip?

If so, IMHO that is quite unpracticable from a developers workflow. One should use Ctrl+S and the file is saved to VCS directly.
Fortunately, PowerPoint supports this: There is a flat file format that embeds everything in an XML file:
![powerPointXml](https://user-images.githubusercontent.com/11966684/80926194-f186ac80-8d95-11ea-83c7-5cb402faa653.png)

Just select "PowerPoint XML" as a file format.

Bonus: You can now edit your complete PPT with a usual text editor (try throwing a lot of images in there! :upside_down_face:):
![text](https://user-images.githubusercontent.com/11966684/80926230-34e11b00-8d96-11ea-971f-a96e6224ac08.png)

<details>
<summary>*…</summary>
<small>Yeah, sorry for Windows 7. It was the best, still sad to see it gone…</small>
</details>

**Comment by aetonsi** [comment]
Yeah, i just unzipped it. My goal was to have multiple files to work on, in order to have as little conflicts as possible.

I didn't know about the single file XML to be honest...

**Comment by rugk** [comment]
Well… an XML file would also avoid the conflicts, as it is not binary.
Multiple files may be more convenient to edit, but – on the other hand – you need to manually extract them. So there is a trade-off to make here…

**Comment by aetonsi** [comment]
Yeah i know.. i included a script to make it possible to pack->edit->unpack the file, but it's surely not practical

**Comment by TomWildenhain** [comment]
Realistically, merging two PPT presentations automatically is going to be a nightmare regardless of how they are stored.  Ultimately if we want to generate PPT files programmatically, there will need to be some type of template file and a python script that programmatically edits it (probably in unpacked or XML form).  Until then, I would recommend making different PPT files for different components and then copying the slides together into a single presentation manually.
